tools/hidl/hidlcomp.cpp

```diff
@@ -4036,29 +4036,18 @@ void EspServInfo::write_esp_binding_ipp()
 
     //
     // Create scaled histogram metric member variables enabled methods
-    // Only output the ifdef if needed
-    bool needIfDef = true;
+    // Always output, even if they are never initialised to prevent problems
+    // where the header is included with inconsistent #defines
```

Comment:
Only comment here is that I would say something like

// where the header is used in a non ESP service context

Not necessary to change.